cmd/antrea-service/main.go

```diff
@@ -0,0 +1,319 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build windows
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/kardianos/service"
+	"golang.org/x/sys/windows"
+)
+
+const (
+	defaultAgentInstallPath = "C:\\antrea\\"
+	systemPath              = "C:\\Windows\\System32\\;C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\;"
+
+	nodeNameEnvKey     = "NODE_NAME"
```

Comment:
Is that Windows Service typically uses env to pass parameters?

Reply:
Theoretically，the parameters could be passed with command line when creating Windows Service, but it requires Windows command utility "sc" to generate the Service. I originally planned to use this wrapper to register the Service directly, and not leverage other commands The parameters provided from the command line don't work in my test with this aussumption.

I could try with "sc" command, if using command line parameter is better than setting envrionment variable. What do you think @jianjuns ?

Reply:
I do not know what is the appropriate to pass parameters to Windows Service. If env variables are simpler, guess it is fine.